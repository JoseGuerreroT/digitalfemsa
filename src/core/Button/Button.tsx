import React from 'react';
import {TouchableOpacity, StyleSheet} from 'react-native';
import {TouchableOpacityProps} from 'react-native';
import Text from '../Text';
type ButtonProps = TouchableOpacityProps & {
  title: string;
  onPress?: () => void;
  size?: 'small' | 'medium';
};

const Button: React.FC<ButtonProps> = ({
  title,
  onPress,
  size = 'medium',
  style,
  ...props
}) => {
  return (
    <TouchableOpacity
      style={[styles.button, style]}
      onPress={onPress}
      {...props}>
      <Text
        weight="bold"
        style={[styles.text, {fontSize: size === 'medium' ? 16 : 12}]}>
        {title}
      </Text>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  button: {
    borderRadius: 10,
    height: 50,
    backgroundColor: '#334FFA',
    justifyContent: 'center',
  },
  text: {
    textAlign: 'center',
    fontSize: 16,
    color: '#fff',
  },
});

export default Button;
