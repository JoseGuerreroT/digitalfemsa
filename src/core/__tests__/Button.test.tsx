import React from 'react';
import {render, fireEvent} from '@testing-library/react-native';
import Button from '../Button';

describe('Button', () => {
  const onPressMock = jest.fn();

  test('01 - renders correctly with default props', () => {
    const {getByText} = render(<Button title="Press me" />);
    const button = getByText('Press me');
    expect(button).toBeDefined();
  });

  test('02 - renders correctly with custom props', () => {
    const {getByText} = render(
      <Button title="Press me" onPress={onPressMock} size="small" />,
    );
    const button = getByText('Press me');
    expect(button).toBeDefined();
  });

  test('03 - calls onPress prop when pressed', () => {
    const {getByText} = render(
      <Button title="Press me" onPress={onPressMock} />,
    );
    const button = getByText('Press me');
    fireEvent.press(button);
    expect(onPressMock).toHaveBeenCalledTimes(1);
  });

  it('04 - renders correctly with size="small"', () => {
    const {toJSON} = render(<Button title="Press me" size="small" />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('05 - renders correctly with size="medium"', () => {
    const {toJSON} = render(<Button title="Press me" size="medium" />);
    expect(toJSON()).toMatchSnapshot();
  });
});
