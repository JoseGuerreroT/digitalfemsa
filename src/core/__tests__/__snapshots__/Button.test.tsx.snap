// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button core component 04 - renders correctly with size="small" 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "#334FFA",
      "borderRadius": 10,
      "height": 50,
      "justifyContent": "center",
      "opacity": 1,
    }
  }
>
  <Text
    style={
      [
        {
          "fontFamily": "Avenir",
        },
        [
          {
            "color": "#fff",
            "fontSize": 16,
            "textAlign": "center",
          },
          {
            "fontSize": 12,
          },
        ],
        {
          "fontWeight": "bold",
        },
      ]
    }
  >
    Press me
  </Text>
</View>
`;

exports[`Button core component 05 - renders correctly with size="medium" 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "#334FFA",
      "borderRadius": 10,
      "height": 50,
      "justifyContent": "center",
      "opacity": 1,
    }
  }
>
  <Text
    style={
      [
        {
          "fontFamily": "Avenir",
        },
        [
          {
            "color": "#fff",
            "fontSize": 16,
            "textAlign": "center",
          },
          {
            "fontSize": 16,
          },
        ],
        {
          "fontWeight": "bold",
        },
      ]
    }
  >
    Press me
  </Text>
</View>
`;
